### PR TITLE
added quiz functionality

### PR DESCRIPTION
**Pull request by DivyanshSareen**

### CODE REVIEW

File: src/rotues/QuizPage.jsx [hunk 1]
```diff
@@ -1,34 +1,50 @@
...
-          </a>
-          <a href='./result.html'>
-            <div className='choice'>Not Ansering the phone</div>
-          </a>
-          <a href='./result.html'>
-            <div className='choice'>Wearing her glasses</div>
-          </a>
+          {question?.options?.map((e) => (
+            <div
+              key={e}
```
Comment (aiman-ansari) [inline src/rotues/QuizPage.jsx:38]: Suggestion: naming could be better here

File: src/App.js [hunk 2]
```diff
@@ -2,14 +2,15 @@ import "./styles/App.css";
 import HomePage from "./rotues/HomePage";
 import { Routes, Route } from "react-router-dom";
 import QuizPage from "./rotues/QuizPage";
+import ResultPage from "./rotues/ResultPage";
```
Comment (HeyNitin) [inline src/App.js:5]: You can have a look at absolute imports for next project, just a suggestion

File: src/App.js [hunk 3]
```diff
@@ -2,14 +2,15 @@ import "./styles/App.css";
...
 
 function App() {
   return (
     <div className='App'>
       <Routes>
         <Route path='/' element={<HomePage />} />
         <Route path='/quiz' element={<QuizPage />} />
-        <Route path='/result' />
```
Comment (HeyNitin) [inline src/App.js:12]: Instead of having routes in main App, you can have a <Routes /> component

File: src/rotues/QuizPage.jsx [hunk 4]
```diff
@@ -1,34 +1,50 @@
...
+    useQuiz();
+
+  const [question, setQuestion] = useState(quiz.questions[quesCounter]);
+  useEffect(() => {
+    if (quesCounter >= quiz.numberOfQuestions) {
+      navigate("/result");
+    } else {
+      setQuestion(quiz.questions[quesCounter]);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (HeyNitin) [inline src/rotues/QuizPage.jsx:17]: why this comment?

**Review by aiman-ansari — APPROVED** [review]
- Variable naming is declarative but some place it could be better .
- Minimal Code and easy to understand.

File: src/App.js (hunk 2 above)
Comment (dev-enforced) [inline src/App.js:5]: Can use barrel exports for the pages later on

**Review by HeyNitin — APPROVED** [review]
left some suggestions, have a look, you can go ahead and merge

File: src/rotues/QuizPage.jsx [hunk 5]
```diff
@@ -1,34 +1,50 @@
...
-          <a href='./result.html'>
-            <div className='choice'>Stealing post-it notes</div>
-          </a>
-          <a href='./result.html'>
-            <div className='choice'>Not Ansering the phone</div>
-          </a>
-          <a href='./result.html'>
-            <div className='choice'>Wearing her glasses</div>
-          </a>
+          {question?.options?.map((e) => (
```
Comment (dev-enforced) [inline src/rotues/QuizPage.jsx:36]: Good use of optional chaining Divyansh

**Review by dev-enforced — APPROVED** [review]
The code looks good and well structured. Good work Divyansh. Do deploy this soon to check for any mistakes/bugs in the live one

File: src/rotues/QuizPage.jsx (hunk 4 above)
Comment (dev-enforced) [reply src/rotues/QuizPage.jsx:17]: So this comment is for disabling the warning/error that comes from eslint package related to the parameters inside the useEffect dependency array.

**Comment by DivyanshSareen** [comment]
Really appreciate all the reviews and suggestions, merging this PR for now but will definitely address these issues in a later commit

**Pull request merged**